In [1]:
import os    
from dotenv  import load_dotenv
from pathlib import Path
import requests
import pandas as pd
pd.options.display.max_colwidth = 600  # Set Full Display
from pprint import pprint

import datetime
from dateutil import relativedelta
from dateutil.relativedelta import relativedelta
from datetime import datetime, timedelta

In [3]:
load_dotenv(dotenv_path = Path("C:/Users/Leon/API_keys/.env"))
auproperty_api=os.getenv('auproperty_vincent2')
type(auproperty_api)

str

# Address Code:

In [4]:
'''
property_search_url = "https://api2.aupropertyreport.com/api/property/search?q=3+17+hillcrest+ave+chadstone+vic+3148"

# GET Request
property_search_request = requests.get(property_search_url,headers=auth,params={"limit":"50"})
property_search_respond = property_search_request.json()
property_search_respond
'''

'\nproperty_search_url = "https://api2.aupropertyreport.com/api/property/search?q=3+17+hillcrest+ave+chadstone+vic+3148"\n\n# GET Request\nproperty_search_request = requests.get(property_search_url,headers=auth,params={"limit":"50"})\nproperty_search_respond = property_search_request.json()\nproperty_search_respond\n'

In [5]:
address_code = "5-cherry-st-glen-waverley-vic-3150"

## First Key:

In [6]:
auth_url = 'https://api2.aupropertyreport.com/api/auth'

auth_response = requests.post(auth_url, data = {
                        'email':"vincentgao076@gmail.com",
                        'api_key':auproperty_api,
                        })
auth_token = auth_response.json() 
access_token = auth_token['data']["token"]
auth = {"Authorization":"Bearer " + access_token}
auth

{'Authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL2FwaTIuYXVwcm9wZXJ0eXJlcG9ydC5jb20vYXBpL2F1dGgiLCJpYXQiOjE2NDg5NzQ2MzUsImV4cCI6MTY0OTA2MTAzNSwibmJmIjoxNjQ4OTc0NjM1LCJqdGkiOiJFc1NUQ0RPY2Y0MkJva1dVIiwic3ViIjozNywicHJ2IjoiNDBhNDZkZjhmNTlkN2E5ZGVkZDc0NjRkZDI2NTNjOGIxNzE2ZTBiZCIsInBsYW4iOm51bGx9.2i_M1avDlfmQXKL1co-gxVzQ7l-KRDLrfWS_uBbZBH4'}

## Second Key:

In [7]:
# Overwrite, initiate second API key directly !!
# property_search_respond={}

In [8]:
property_search_url = "https://api2.aupropertyreport.com/api/property/search?q=3+17+hillcrest+ave+chadstone+vic+3148"

# GET Request
property_search_request = requests.get(property_search_url,headers=auth,params={"limit":"50"})
property_search_respond = property_search_request.json()
property_search_respond

{'message': 'You have exceeded your rate limit.', 'status_code': 429}

Exceed: {'message': 'You have exceeded your rate limit.', 'status_code': 429}

Normal:  [{'address': '3/17 Hillcrest Avenue, Chadstone Vic 3148',
  'code': 'unit-3-17-hillcrest-ave-chadstone-vic-3148',
  'state': 'VIC',
  'suburb': 'Chadstone',
  'postcode': '3148'}]

In [9]:
# 由于 address code search 的 respond 都是 list, 即使是空的，或者多个的，用来check rate-limit
if isinstance(property_search_respond, dict):

    auproperty_api=os.getenv('auproperty_vincent2')

    auth_url = 'https://api2.aupropertyreport.com/api/auth'

    auth_response = requests.post(auth_url, data = {
                        'email':"leon20130301@hotmail.com",
                        'api_key':os.getenv('auproperty_vincent'),
                        })
    auth_token = auth_response.json() 
    access_token = auth_token['data']["token"]
    auth = {"Authorization":"Bearer " + access_token}
    print(auth) # 如果没有任何显示，则说明没有使用这个 API key

{'Authorization': 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL2FwaTIuYXVwcm9wZXJ0eXJlcG9ydC5jb20vYXBpL2F1dGgiLCJpYXQiOjE2NDg5NzQ2MzgsImV4cCI6MTY0OTA2MTAzOCwibmJmIjoxNjQ4OTc0NjM4LCJqdGkiOiJUVG0xdzRZU2kxbkVDQ0twIiwic3ViIjozNiwicHJ2IjoiNDBhNDZkZjhmNTlkN2E5ZGVkZDc0NjRkZDI2NTNjOGIxNzE2ZTBiZCIsInBsYW4iOiJtb250aGx5X3BsYW5fYV9zbyJ9.21g5qQJuMG_3opyDMt74sQUUS1uluDxXugm2InKT-AU'}


# Similar Sold: 過去一年 多個區域

In [10]:
# use r -- to avoid unicode error !
filesaved_similarsold_path=Path(rf'C:\Users\LEON\test\domain\Similar_Sold_for_{address_code}.xlsx')
filesaved_similarsold_df = pd.read_excel(filesaved_similarsold_path, parse_dates=True)
print(filesaved_similarsold_df.info())
filesaved_similarsold_df.tail(1)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 740 entries, 0 to 739
Data columns (total 19 columns):
state                 740 non-null object
address               740 non-null object
suburb_code           740 non-null object
suburb_name           740 non-null object
postcode              740 non-null int64
image                 740 non-null object
type                  740 non-null object
bed                   740 non-null int64
bath                  740 non-null int64
car                   740 non-null int64
lat                   740 non-null float64
lng                   740 non-null float64
land                  616 non-null float64
sold_date             740 non-null object
sold_price            740 non-null int64
gnaf_pid              733 non-null object
pu_id                 740 non-null int64
agent_code            739 non-null object
sold_price_display    740 non-null object
dtypes: float64(3), int64(6), object(10)
memory usage: 110.0+ KB
None


state             address              suburb_code     suburb_name  \
739   Vic  14 Charlton Street  mount-waverley-vic-3149  Mount Waverley   

     postcode  \
739      3149   

                                                                 image   type  \
739  https://farm66.static.flickr.com/65535/51024194568_1f13392163.jpg  house   

     bed  bath  car        lat         lng   land   sold_date  sold_price  \
739    4     2    3 -37.882934  145.125983  724.0  2021-03-01     1850000   

           gnaf_pid        pu_id agent_code       sold_price_display  
739  GAVIC421246113  39736557218  VIC247287  $1,817,000 - $1,903,000

非常注意：

不會保存原來數據！只要最新數據！所以時間範圍也是新的！

In [11]:
similar_sold_url = f"https://api2.aupropertyreport.com/api/property/{address_code}/similar-sold"
# GET Request
similar_sold_request_1 = requests.get(similar_sold_url,headers=auth,params={"limit":"50"})
similar_sold_respond_1 = similar_sold_request_1.json()

if "message" not in similar_sold_respond_1:

    # get total page number:
    similar_sold_pagesnum = similar_sold_respond_1["data"]["similar_sold"]["meta"]["pagination"]["total_pages"]
    # save the first call API data to dataframe:
    similar_sold_df = pd.DataFrame(similar_sold_respond_1["data"]["similar_sold"]['data'])

    for page_num in range(2,similar_sold_pagesnum+1): # first entry saved, range not inclusive last
        similar_sold_url_2 = f"https://api2.aupropertyreport.com/api/property/{address_code}/similar-sold?page={page_num}"
        # GET Request
        similar_sold_request_2 = requests.get(similar_sold_url_2,headers=auth,params={"limit":"50"})
        similar_sold_respond_2 = similar_sold_request_2.json()

        if "message" not in similar_sold_respond_2:
            similar_sold_temp_df = pd.DataFrame(similar_sold_respond_2["data"]["similar_sold"]['data'])
            similar_sold_df = similar_sold_df.append(similar_sold_temp_df)

        else:
            print("ERROR IN RESPOND")
            break    

    if "message" not in similar_sold_respond_2:       
        similar_sold_df = similar_sold_df.drop_duplicates(subset=["address","sold_date"],keep="last")
        similar_sold_updated_df = similar_sold_df.append(similar_sold_df)
        similar_sold_updated_df = similar_sold_updated_df.drop_duplicates(subset=["address","sold_date"],keep="last")

        similar_sold_num_duplication = similar_sold_updated_df.duplicated(["address","sold_date"]).sum()

        # create excel writer object
        writer=pd.ExcelWriter(f'Similar_Sold_for_{address_code}.xlsx')
        # write dataframe to excel
        similar_sold_updated_df.to_excel(writer, index=False)
        # save the excel
        writer.save()
        print('DataFrame is written successfully to Excel File.')
        print(similar_sold_updated_df.info())
        print("---------------------------------------------")
        print(f"Updated Entries: {len(similar_sold_updated_df)-len(filesaved_similarsold_df)}")
        print("---------------------------------------------")
        print(f"Check for duplication: {similar_sold_num_duplication}")
        
    else: 
        print("ERROR IN RESPOND")
else:   
    print("ERROR IN RESPOND")

DataFrame is written successfully to Excel File.
<class 'pandas.core.frame.DataFrame'>
Int64Index: 658 entries, 0 to 8
Data columns (total 19 columns):
state                 658 non-null object
address               658 non-null object
suburb_code           658 non-null object
suburb_name           658 non-null object
postcode              658 non-null object
image                 658 non-null object
type                  658 non-null object
bed                   658 non-null int64
bath                  658 non-null int64
car                   658 non-null int64
lat                   658 non-null float64
lng                   658 non-null float64
land                  658 non-null object
sold_date             658 non-null object
sold_price            658 non-null int64
gnaf_pid              658 non-null object
pu_id                 658 non-null int64
agent_code            657 non-null object
sold_price_display    658 non-null object
dtypes: float64(2), int64(5), object(12)
memory usage

### Similar in Particular Surburb with Specific Conditions:


非常注意：

如果限定很具體，那麽可能出現 比如 滿足 5B 3B 2C 同區域房子，是去年的，那麽不能反應現在的價格。

#### Clean Land to Int:

In [12]:
# Match Conditions:
suburb1="Glen Waverley"
suburb2="Mount Waverley"
bed = 5
bath = 3
car = 2
land = 650+100

In [27]:
similar_sold_updated_df_cleaned = similar_sold_updated_df
similar_sold_updated_df_cleaned ["land"] = pd.to_numeric(similar_sold_updated_df["land"]).fillna(0)

In [28]:
# Check land data type:
type(similar_sold_updated_df_cleaned .iloc[5]["land"])

numpy.float64

轉換 Datatime:

In [16]:
similar_sold_updated_df_cleaned["sold_date"] = pd.to_datetime(similar_sold_updated_df_cleaned["sold_date"])
similar_sold_updated_df_cleaned.head(1)

state               address            suburb_code   suburb_name postcode  \
0   VIC  54 Lindisfarne Drive  burwood-east-vic-3151  Burwood East     3151   

                                                               image   type  \
0  https://farm66.static.flickr.com/65535/51965658347_d9cc308200.jpg  house   

   bed  bath  car        lat         lng   land  sold_date  sold_price  \
0    5     2    4 -37.855674  145.150532  542.0 2022-03-28     1365000   

         gnaf_pid        pu_id agent_code       sold_price_display  
0  GAVIC420796943  39732772180  VIC210199  $1,328,000 - $1,392,000

In [29]:
# Check sold_date type:
type(similar_sold_updated_df_cleaned.iloc[5]["sold_date"])

pandas._libs.tslibs.timestamps.Timestamp

篩選，時間到現在一年内的：

In [30]:
now_date = datetime.now()
now_date

datetime.datetime(2022, 4, 3, 18, 41, 9, 272520)

In [31]:
# 半年前：
halfyrsago = now_date - relativedelta(years=1)
benckmark=datetime.strftime(halfyrsago,"%Y-%m-%d")
# benckmark=datetime.strptime(benckmark,"%Y-%m-%d")
benckmark

'2021-04-03'

選擇 1 年内 的數據：

In [32]:
similar_sold_updated_df_cleaned = similar_sold_updated_df_cleaned[similar_sold_updated_df_cleaned["sold_date"]>benckmark]
similar_sold_updated_df_cleaned = similar_sold_updated_df_cleaned.sort_values(by=["sold_date"],ascending = False)
similar_sold_updated_df_cleaned

state               address              suburb_code     suburb_name  \
46   VIC        2 Cohuna Court    burwood-east-vic-3151    Burwood East   
0    VIC           2 Ian Grove         burwood-vic-3125         Burwood   
49   Vic    24 Summit Crescent   glen-waverley-vic-3150   Glen Waverley   
48   VIC       26 Ashmore Road     forest-hill-vic-3131     Forest Hill   
47   VIC     377 Waverley Road  mount-waverley-vic-3149  Mount Waverley   
..   ...                   ...                      ...             ...   
5    VIC        24 Crow Street    burwood-east-vic-3151    Burwood East   
4    Vic            2 Brent St   glen-waverley-vic-3150   Glen Waverley   
7    VIC  242  Gallaghers Road   glen-waverley-vic-3150   Glen Waverley   
44   VIC      15 Tobias Avenue   glen-waverley-vic-3150   Glen Waverley   
45   VIC     35 Kinnoull Grove   glen-waverley-vic-3150   Glen Waverley   

   postcode  \
46     3151   
0      3125   
49     3150   
48     3131   
47     3149   
..      ...   
5      3151   
4      3150   
7      3150   
44     3150   
45     3150   

                                                                image   type  \
46  https://farm66.static.flickr.com/65535/51978734814_7cf8766c92.jpg  house   
0   https://farm66.static.flickr.com/65535/51977105512_98246e5acd.jpg  house   
49  https://farm66.static.flickr.com/65535/51978082280_cc2a18b24f.jpg  house   
48  https://farm66.static.flickr.com/65535/51976374037_3fed18fb0a.jpg  house   
47  https://farm66.static.flickr.com/65535/51977566009_9dee1f89d0.jpg  house   
..                                                                ...    ...   
5   https://farm66.static.flickr.com/65535/51119314485_028184bba5.jpg  house   
4   https://farm66.static.flickr.com/65535/51104164722_0eff0ef6b0.jpg  house   
7   https://farm66.static.flickr.com/65535/51117883397_201e140e1b.jpg  house   
44  https://farm66.static.flickr.com/65535/51104495053_4aba6b5d0b.jpg  house   
45  https://farm66.static.flickr.com/65535/51112084217_068ce1a053.jpg  house   

    bed  bath  car        lat         lng   land  sold_date  sold_price  \
46    4     2    2 -37.858624  145.153607  565.0 2022-04-03     1305000   
0     4     2    1 -37.855808  145.120340  579.0 2022-04-02     1321000   
49    4     2    2 -37.888410  145.172183  694.0 2022-04-02     1706500   
48    4     3    2 -37.840395  145.182131  586.0 2022-04-02     1280000   
47    4     3    4 -37.884766  145.129530  806.0 2022-04-02     1502000   
..  ...   ...  ...        ...         ...    ...        ...         ...   
5     4     2    1 -37.854138  145.136568    0.0 2021-04-09     1191000   
4     4     2    1 -37.875807  145.167844  783.0 2021-04-09     1750000   
7     4     2    2 -37.880465  145.178257  712.0 2021-04-08     1600000   
44    5     3    2 -37.887262  145.168470  736.0 2021-04-06     1790000   
45    5     2    2 -37.880653  145.155035  673.0 2021-04-06     1800000   

          gnaf_pid        pu_id agent_code       sold_price_display  
46                  39768196751  VIC212651  $1,269,000 - $1,331,000  
0                   39766251451  VIC181900  $1,297,000 - $1,359,000  
49  GAVIC420971047  39783440851  VIC318247  $1,675,000 - $1,755,000  
48  GAVIC419586993  39785199651  VIC113859  $1,257,000 - $1,319,000  
47  GAVIC420104558  39785842551  VIC358007  $1,475,000 - $1,545,000  
..             ...          ...        ...                      ...  
5   GAVIC421088307  39751323328  VIC167454  $1,168,000 - $1,226,000  
4   GAVIC421234292  39772454608  VIC250940  $1,704,000 - $1,786,000  
7   GAVIC419772385  39755544259  VIC158946  $1,558,000 - $1,632,000  
44  GAVIC421927062  39732531439  VIC254295  $1,758,000 - $1,842,000  
45  GAVIC421832127  39730848218  VIC320293  $1,770,000 - $1,854,000  

[657 rows x 19 columns]

In [33]:
similar_sold_updated_df2=similar_sold_updated_df_cleaned[((similar_sold_updated_df_cleaned["suburb_name"]==f"{suburb1}") | \
                        (similar_sold_updated_df_cleaned["suburb_name"]==f"{suburb2}")) & \
                             (similar_sold_updated_df_cleaned["bed"]==bed) & \
                                (similar_sold_updated_df_cleaned["bath"]==bath) & \
                                    (similar_sold_updated_df_cleaned["car"]==car) & \
                                        (similar_sold_updated_df_cleaned["land"]<=land) & \
                                            (similar_sold_updated_df_cleaned["land"]> 0) \
                                    ]

similar_sold_updated_df2_sorted = similar_sold_updated_df2.sort_values("sold_price",ascending=False)

# create excel writer object
writer=pd.ExcelWriter(f'Similar_Sold_for_{address_code}_Specific.xlsx')
# write dataframe to excel
similar_sold_updated_df2_sorted.to_excel(writer, index=False)
# save the excel
writer.save()
print('DataFrame is written successfully to Excel File.')
print(similar_sold_updated_df2.info())


DataFrame is written successfully to Excel File.
<class 'pandas.core.frame.DataFrame'>
Int64Index: 21 entries, 28 to 44
Data columns (total 19 columns):
state                 21 non-null object
address               21 non-null object
suburb_code           21 non-null object
suburb_name           21 non-null object
postcode              21 non-null object
image                 21 non-null object
type                  21 non-null object
bed                   21 non-null int64
bath                  21 non-null int64
car                   21 non-null int64
lat                   21 non-null float64
lng                   21 non-null float64
land                  21 non-null float64
sold_date             21 non-null datetime64[ns]
sold_price            21 non-null int64
gnaf_pid              21 non-null object
pu_id                 21 non-null int64
agent_code            21 non-null object
sold_price_display    21 non-null object
dtypes: datetime64[ns](1), float64(3), int64(5), object(10)
me

## Check Duplication:

PS: 純粹重複

In [14]:
# similar_sold_updated_df.duplicated(["address","sold_date"]).sum()

739

In [15]:
# similar_sold_updated_df[similar_sold_updated_df.duplicated(["address","sold_date"])]

state               address               suburb_code      suburb_name  \
25   VIC     26 Andrews Street          burwood-vic-3125          Burwood   
0    VIC  54 Lindisfarne Drive     burwood-east-vic-3151     Burwood East   
1    VIC   323 Gallaghers Road    glen-waverley-vic-3150    Glen Waverley   
2    VIC      15 Jasmine Court  blackburn-south-vic-3130  Blackburn South   
3    VIC       28 Marianne Way   mount-waverley-vic-3149   Mount Waverley   
..   ...                   ...                       ...              ...   
33   VIC     21 Narmara Street     burwood-east-vic-3151     Burwood East   
34   VIC    36 Faulkner Street      forest-hill-vic-3131      Forest Hill   
35   VIC       21 Smyth Street   mount-waverley-vic-3149   Mount Waverley   
36   Vic     15 Bulkara Avenue      forest-hill-vic-3131      Forest Hill   
37   Vic    14 Charlton Street   mount-waverley-vic-3149   Mount Waverley   

   postcode  \
25     3125   
0      3151   
1      3150   
2      3130   
3      3149   
..      ...   
33     3151   
34     3131   
35     3149   
36     3131   
37     3149   

                                                                                                                                                                                                                                                       image  \
25                                                                                                                                                                                         https://farm66.static.flickr.com/65535/51156759136_d54c2aa7b7.jpg   
0                                                                                                                                                                                          https://farm66.static.flickr.com/65535/51965658347_d9cc308200.jpg   
1   https://i7.auhouseprices.com/img/MmRXVnpmWkMwSXdtdjBPN2NEZUpkWk1SK1YrcDB2ajNUd2MwaTM5SzhQVFhycWZ4N2tPSnJUYXRFZkxLOHpSUzJvOHBhVUJCaDZiZjVIMkJ2ajB0WExaNVFqVEJSVmpHVjh5MXFlYTE1cWFaWUMyWC9SS2tFVkY0NVdWd2ZYdVdXbSs3Vm9CaTlYbTZmRVBuVXlYcjZ3PT0/preview.jpg   
2                                                                                                                                                                                          https://farm66.static.flickr.com/65535/51962778885_fc06169c37.jpg   
3                                                                                                                                                                                          https://farm66.static.flickr.com/65535/51963063750_516f9074fe.jpg   
..                                                                                                                                                                                                                                                       ...   
33                                                                                                                                                                                         https://farm66.static.flickr.com/65535/51028077287_1d0c7122e6.jpg   
34                                                                                                                                                                                         https://farm66.static.flickr.com/65535/50991846077_0e53ccd62c.jpg   
35                                                                                                                                                                                         https://farm66.static.flickr.com/65535/50992084863_8e292a207f.jpg   
36                                                                                                                                                                                         https://farm66.static.flickr.com/65535/51005067807_a6f4074a51.jpg   
37                                                                               

# Property Sold Trend Data：类似房數，半径2km 根據每年 統計 月中价位

## Initialization: Loading saved data

In [21]:
# use r -- to avoid unicode error !
filesaved_soldtrend_path=Path(rf'C:\Users\LEON\test\domain\Sold_Trend_{address_code}.xlsx')
filesaved_soldtrend_df = pd.read_excel(filesaved_soldtrend_path, parse_dates=True)
print(filesaved_soldtrend_df.info())
filesaved_soldtrend_df.tail(1)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13 entries, 0 to 12
Data columns (total 22 columns):
price_sum             13 non-null int64
price_avg             13 non-null int64
price_median          13 non-null int64
standard_deviation    13 non-null int64
price_min             13 non-null int64
price_10th            13 non-null int64
price_20th            13 non-null int64
price_25th            13 non-null int64
price_30th            13 non-null int64
price_40th            13 non-null int64
price_50th            13 non-null int64
price_60th            13 non-null int64
price_70th            13 non-null int64
price_75th            13 non-null int64
price_80th            13 non-null int64
price_90th            13 non-null int64
price_max             13 non-null int64
num_all               13 non-null int64
num_all_with_price    13 non-null int64
start_date            13 non-null object
end_date              13 non-null object
growth                12 non-null float64
dtypes: float

price_sum  price_avg  price_median  standard_deviation  price_min  \
12  221766052    2132366       1912500              998652    1021000   

    price_10th  price_20th  price_25th  price_30th  price_40th  ...  \
12     1409500     1547400     1648000     1674950     1800000  ...   

    price_70th  price_75th  price_80th  price_90th  price_max  num_all  \
12     2360200     2430000     2574000     2989200   10100000      147   

    num_all_with_price  start_date    end_date  growth  
12                 104  2021-03-01  2022-02-28 -0.0107  

[1 rows x 22 columns]

In [23]:
property_sold_trend_url = f"https://api2.aupropertyreport.com/api/property/{address_code}/sold-trend-data"

# GET Request
property_sold_trend_request = requests.get(property_sold_trend_url,headers=auth,params={"limit":"50"})
property_sold_trend_respond = property_sold_trend_request.json()

if "message" not in property_sold_trend_respond:

    # save the first call API data to dataframe:
    property_sold_trend_df = pd.DataFrame.from_dict(property_sold_trend_respond["data"]["sold_trend_data"]['data']["trend_list"],orient='index')
    property_sold_updated_df = filesaved_soldtrend_df.append(property_sold_trend_df)

    property_sold_updated_df = property_sold_updated_df.drop_duplicates(subset=["start_date","end_date"],keep="last")
    property_sold_num_duplication = property_sold_updated_df.duplicated(["start_date","end_date"]).sum()

    # create excel writer object
    writer=pd.ExcelWriter(f'Sold_Trend_{address_code}.xlsx')
    # write dataframe to excel
    property_sold_trend_df.to_excel(writer, index=False)
    # save the excel
    writer.save()
    print('DataFrame is written successfully to Excel File.')
    print("---------------------------------------------")
    print(property_sold_trend_df.info())
    print("---------------------------------------------")
    print(f"Check for duplication: {property_sold_num_duplication}")

else:   
    print("ERROR IN RESPOND")

DataFrame is written successfully to Excel File.
---------------------------------------------
<class 'pandas.core.frame.DataFrame'>
Index: 13 entries, 2021-02 to 2022-02
Data columns (total 22 columns):
price_sum             13 non-null int64
price_avg             13 non-null int64
price_median          13 non-null int64
standard_deviation    13 non-null int64
price_min             13 non-null int64
price_10th            13 non-null int64
price_20th            13 non-null int64
price_25th            13 non-null int64
price_30th            13 non-null int64
price_40th            13 non-null int64
price_50th            13 non-null int64
price_60th            13 non-null int64
price_70th            13 non-null int64
price_75th            13 non-null int64
price_80th            13 non-null int64
price_90th            13 non-null int64
price_max             13 non-null int64
num_all               13 non-null int64
num_all_with_price    13 non-null int64
start_date            13 non-null ob

# Property History:

In [9]:
property_history_url = f"https://api2.aupropertyreport.com/api/property/{address_code}/history"

# GET Request
property_history_request = requests.get(property_history_url,headers=auth,params={"limit":"50"})
property_history_respond=property_history_request.json()

if "message" not in property_history_request: 
    property_fullinfor_df = pd.DataFrame.from_dict(property_history_respond["data"])
else:   
    print("ERROR IN RESPOND")    

In [10]:
for each in property_fullinfor_df["history"]:
    pprint(each)

[{'agency': 'Llc Real Estate - Mount Waverley',
  'date': '2017-06-02',
  'event': 'Sold',
  'price': 1800000},
 {'date': '2002-04-27', 'event': 'Sold', 'price': 322000},
 {'date': '1990-04-24', 'event': 'Sold', 'price': 173000},
 {'date': '1982-05-08', 'event': 'Sold', 'price': 53600},
 {'date': '1981-05-14', 'event': 'Sold', 'price': 61000}]


In [32]:
# property_fullinfor_df

In [17]:
property_table = property_fullinfor_df[["address_code","bed","bath","car","land","building","year_built"]].reset_index(drop=True)
property_table



address_code  bed  bath  car land  building  \
0  5-cherry-st-glen-waverley-vic-3150    5     3    2  650       524   

   year_built  
0        2009

In [33]:
# create excel writer object
writer=pd.ExcelWriter(f'Property_Details_{address_code}.xlsx')
# write dataframe to excel
property_table.to_excel(writer, index=False)
# save the excel
writer.save()
print('DataFrame is written successfully to Excel File.')

DataFrame is written successfully to Excel File.


In [31]:
property_history_df=pd.DataFrame()
for each in property_fullinfor_df["history"]:
    temp = pd.DataFrame(each)
    property_history_df=property_history_df.append(temp)
property_history_df

event        date    price                            agency
0  Sold  2017-06-02  1800000  Llc Real Estate - Mount Waverley
1  Sold  2002-04-27   322000                               NaN
2  Sold  1990-04-24   173000                               NaN
3  Sold  1982-05-08    53600                               NaN
4  Sold  1981-05-14    61000                               NaN

In [34]:
# create excel writer object
writer=pd.ExcelWriter(f'Property_History_{address_code}.xlsx')
# write dataframe to excel
property_history_df.to_excel(writer, index=False)
# save the excel
writer.save()
print('DataFrame is written successfully to Excel File.')

DataFrame is written successfully to Excel File.
